### ISSUE TITLE
Problem with History junction (Win 7)

### ISSUE THREAD

**Issue by Geist-zz** [AlisterH/gwhere-catalog#5]
Gwhere stops scanning partition with an error "Folder doesn't exist" on encountering Historia (History) junction under Windows 7.

From dir output on Win 7 system partition (`dir c:\Users\User_name\AppData\Local /ahs`; target folder exists):
`<JUNCTION>     Historia [C:\Users\User_name\AppData\Local\Microsoft\Windows\History]`
The exact same output is for the Vista partition (disk from laptop, connected internally as I:, dir made with an i:\...), same problem.

![junction](https://github.com/AlisterH/gwhere-catalog/assets/749062/b71d3054-b12d-43dd-9859-240a0954036d)

Also under Win 7 export to csv crashes the program.

Could be something specific to that Windows 7 desktop.

Under Windows 11 scanning its system partition, and exporting to csv doesn't cause the problems above.

Possible solution to scan breaking: show warning about missing folder(s), but let the scan continue instead of stopping.

PS. After stopping the scan on error, the catalog is considered as modified (asking to save on exit), despite that nothing was added to it.

**Comment by AlisterH** [comment]
Thanks.  Are you able to build a debug version and get a backtrace for that crash?

Note that there is a version of gwhere [here](https://github.com/wdlkmpx/gwhere/commits/master) which has been maintained much more recently - I'd like to do a pull request to merge my changes there - it just isn't a priority so I never get around to it.
It is probably worth filing a ticket there too, assuming that version has the same problems.

**Comment by Geist-zz** [comment]
Not really, unless I get help with setting up the development environment under Windows (Cygwin?). Or preferably someone can point me to a small (preferable below 8/16 GB) distribution of Linux that can run from USB/DVD and is capable of compiling the program (well, probably most distributions are, I just don't want to experiment with it too much. I may try it anyway just to check if the problem also appear under Linux).
I'm using the last official Windows release of gwhere (downloaded from gwhere.org through web.archive.org).
It's not a crash, the disk scanning just stops with that error (without adding anything to a catalog).

The other project from what I see also requires compiling (and Linux).

**Comment by AlisterH** [comment]
>It's not a crash, the disk scanning just stops with that error (without adding anything to a catalog).

But you also said this:
> Also under Win 7 export to csv crashes the program.
> 
> Could be something specific to that Windows 7 desktop.

The crash is what needs debugging, so no need to build a debug version under Linux or anything, unless we know it also crashes in Linux.

> unless I get help with setting up the development environment under Windows (Cygwin?)

No need for Cygwin - it'll be msys I presume.

> I'm using the last official Windows release of gwhere (downloaded from gwhere.org through web.archive.org).

I've just done some testing with the same release.  I couldn't get it to crash (Windows 10 here), but did notice a minor issue - if I try to overwrite an existing csv which is read-only it thinks it is successful.
